### DISCUSSION TITLE
Support for semantic versioning in workflows

### DISCUSSION THREAD

**Discussion by RussellRC** [Netflix/conductor#2225] [Ideas]
Currently you can version your workflow definitions with an integer number, which is nice because you can have multiple versions for the same workflow.

However, as with any app, sometimes you have to patch an existing production-in-use version, while you already have the next minor or major version in the works, or even already running in a non-production environment, in which case having an incremental integer version number is not very helpful.

I think this would be a really useful feature.

**Comment by apanicker-nflx** [comment]
HI @RussellRC, while conceptually the idea of semantic versioning in workflows would help, this would lead to increased complexity for both the operator and user of Conductor. From the development POV, extra implementation would be needed to support various implementation modes of a workflow (run newest patch for a given minor version/newest minor version for a given major version, etc) From the operator POV, removing and managing workflow definitions would require extra coordinated effort. From a user POV, it increases the level of indulgence to identify what version of a workflow needs to be used. 

Weighing the complexity increase against the potential gain, this does not seem to add a lot of value. Please let us know if you think otherwise.

**Reply by RussellRC** [reply]
Thanks a lot for the reply @apanicker-nflx, I totally understand your point.

As a workaround, I figured I can use tens or hundreds as an initial "production-ready" version (e.g. 100, 200, etc.), and then patch using increments of one (e.g. 101, 102, 201, etc.)

**Reply by apanicker-nflx** [reply]
That is a good workaround for your use-case while reusing the current versioning scheme.
